tests/test_core.py

```diff
@@ -3481,7 +3481,7 @@ def indir(name):
         return 0;
       }
       ''')
-    self.do_runf('main.c', 'a: loaded\nb: loaded\na: loaded\n')
+    self.do_runf('main.c', 'a: loaded\nb: loaded\n')
 
```

Comment:
This is because we don't actually even unload libraries.  We now use dlclose from musl which ignores the refcount on a loaded library.     See the PR description for more on this.

For what its worth our old vesion of `dlclose` didn't really much of a closing.  I didn't run destructors and it didn't free table space of TLS memory or unload the actual module in any way.. its only stopped tracking it.. which is think it wrong/worse.

Reply:
I side effect of never uploading libraries is that we never re-load them and we never re-run the static constructors.. which is that this test here was verifiying.

Reply:
I could implement reference counting and a form of "unloading" in the native version but its seems unnecessary (given than musl doesn't do it).. and maybe heard/impossible to implement any better than we were doing in the JS version.

Reply:
Doesn't seem important to do more work, I guess. But I'd like to understand why before it loaded the module `a` twice and now only once. I don't follow from your explanation what change caused that? (if we never unloaded libraries, why did we load them again later?)

Reply:
Prior to this change we were reference counting the loaded libraries.    When `dlclose` was called on a library the refcount would be decremented, and when the last handle closed the refcount would reach zero.   When the recount reached zero we could "forget" about the loaded library.. and then a future call to `dlopen` would re-instantiate the module and re-run the static initializers and we would end up with a second copy of the same module loaded in the table and memory space.

The way this can be observed is via the re-running on the static initializers, as in this test case.    However, there are no guarantees as far as I can tell about if/when a library is actually unloaded, so simply never unloading it seems like a reasonable approach, and one that musl seems to have take already.    

The previous approach we were taking of claiming to to have unloaded a module and the loading it for a second time seems strictly worse.

From the man page for dlclose:

```
       A  successful return from dlclose() does not guarantee that the symbols associated with handle
       are removed from the caller's address space.  In addition to  references  resulting  from  ex‐
       plicit dlopen() calls, a shared object may have been implicitly loaded (and reference counted)
       because of dependencies in other shared objects.  Only when all references have been  released
       can the shared object be removed from the address space.
```

Reply:
I see, thanks, then we were doing more before. But we think it's best to not do that work. sgtm.